coreclient/src/conversations/status.rs

```diff
@@ -0,0 +1,226 @@
+// SPDX-FileCopyrightText: 2025 Phoenix R&D GmbH <hello@phnx.im>
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use enumset::EnumSetType;
+use mimi_content::MessageStatus;
+
+pub(crate) struct StatusRecord {}
+
+#[derive(Debug, EnumSetType)]
+pub enum MessageStatusBit {
+    Unread = 0,
+    Delivered = 1,
+    Read = 2,
+    Expired = 3,
+    Deleted = 4,
+    Hidden = 5,
+    Error = 6,
+}
+
+impl TryFrom<MessageStatus> for MessageStatusBit {
+    type Error = ();
+
+    fn try_from(value: MessageStatus) -> Result<Self, ()> {
+        match value {
+            MessageStatus::Unread => Ok(Self::Unread),
+            MessageStatus::Delivered => Ok(Self::Delivered),
+            MessageStatus::Read => Ok(Self::Read),
+            MessageStatus::Expired => Ok(Self::Expired),
+            MessageStatus::Deleted => Ok(Self::Deleted),
+            MessageStatus::Hidden => Ok(Self::Hidden),
+            MessageStatus::Error => Ok(Self::Error),
+            MessageStatus::Custom(_) => Err(()),
+        }
+    }
+}
+
+impl From<MessageStatusBit> for MessageStatus {
+    fn from(value: MessageStatusBit) -> Self {
+        match value {
+            MessageStatusBit::Unread => Self::Unread,
+            MessageStatusBit::Delivered => Self::Delivered,
+            MessageStatusBit::Read => Self::Read,
+            MessageStatusBit::Expired => Self::Expired,
+            MessageStatusBit::Deleted => Self::Deleted,
+            MessageStatusBit::Hidden => Self::Hidden,
+            MessageStatusBit::Error => Self::Error,
+        }
+    }
+}
+
+mod persistence {
+    use std::collections::HashMap;
+
+    use enumset::EnumSet;
+    use mimi_content::{MessageStatusReport, PerMessageStatus};
+    use phnxcommon::{identifiers::UserId, time::TimeStamp};
+    use sqlx::{SqliteTransaction, query, query_scalar};
+    use tracing::warn;
+
+    use crate::{ConversationMessageId, store::StoreNotifier};
+
+    use super::*;
+
+    impl StatusRecord {
+        pub(crate) async fn store_report(
+            txn: &mut SqliteTransaction<'_>,
+            notifier: &mut StoreNotifier,
+            sender: &UserId,
```

Comment:
Would it make sense to put this data into the actual StatusRecord struct? If it's just an empty struct we could just turn this into a standalone function.

Reply:
Done. Moved the fields in the struct.